include/soci/logger.h

```diff
@@ -30,6 +30,11 @@ class SOCI_DECL logger_impl
     // Called to indicate that a new query is about to be executed.
     virtual void start_query(std::string const & query) = 0;
 
+    // Called to log a parameter that is bound to the currently active query
+    virtual void add_query_parameter(std::string name, std::string value) = 0;
```

Comment:
I wonder if this really should be pure virtual. Both not implementing it at all (i.e. leaving it empty by default) and just storing the values (as standard logger does by default) look reasonable default implementations.

It's also somewhat inconsistent with `get_last_query_with_context()` not being pure virtual.

Reply:
I'll make storing the parameters the default :+1: